src/operator/contrib/proposal.cu

```diff
@@ -553,10 +553,10 @@ class ProposalGPUOp : public Operator{
                                 cudaMemcpyHostToDevice));
 
     // copy results after nms
-    dimGrid.x = (rpn_post_nms_top_n + kMaxThreadsPerBlock - 1) / kMaxThreadsPerBlock;
+    dimGrid.x = (param_.rpn_post_nms_top_n + kMaxThreadsPerBlock - 1) / kMaxThreadsPerBlock;
```

Comment:
Should `param_.rpn_post_nms_top_n` be used here? Or `rpn_post_nms_top_n` computed here originally https://github.com/apache/incubator-mxnet/blob/master/src/operator/contrib/proposal.cu#L438 ? What's the bug in the original version? cc @precedenceguo

Reply:
@ZiyueHuang If `count_anchor` < `param_.rpn_post_nms_top_n`, 
`rpn_post_nms_top_n` will be `count_anchor`, 
and `rpn_post_nms_top_n` will be less than `param_.rpn_post_nms_top_n`.

https://github.com/apache/incubator-mxnet/blob/master/src/operator/contrib/proposal-inl.h#L119
It shows the `batch_size` of the output is `param_.rpn_post_nms_top_n` instead of `rpn_post_nms_top_n`.

Reply:
You are right. Elements whose index is larger than `rpn_post_nms_top_n` should be assigned. The values are copied from the valid anchors, just to satisfy the output size requirement.

Reply:
@precedenceguo Thank you!

Reply:
Yes, it should; but throwing out would require a new array. Instead I set the predicted confidence to be -1, same as the boxes whose centers are out of the input image boundary. They would be ranked to the bottom for NMS so a proper threshold did not produce significant problems.

Reply:
Thank you! 
The invalid anchors would be ranked to the bottom for NMS, but the threshold is for overlap between two anchors rather than score.
Should we add a condition that `score != -1` in NMS?

https://github.com/apache/incubator-mxnet/blob/master/src/operator/contrib/proposal.cc#L259

Reply:
I think it is a good idea but I did not find significant issue of the invalid anchors.